frontend/src/v5/ui/routes/viewer/tickets/ticketsForm/properties/coordsProperty/coordsProperty.helpers.ts

```diff
@@ -62,37 +62,60 @@ const getColorFromMapping = (ticket: ITicket, pinMapping: IPinColorMapping) => {
 };
 
 export const getPinColorHex = (name: string, template: ITemplate, ticket: ITicket) => {
-	const pinSchema = getPinSchema(name, template);
-	if (typeof pinSchema === 'boolean') return DEFAULT_COLOR; // if default pin with no colouring set
-	if (isArray(pinSchema?.color)) return rgbToHex(pinSchema.color); // a custom colour is set, no mapping
-	if (isObject(pinSchema?.color)) return getColorFromMapping(ticket, pinSchema.color); // a custom colour is set with mapping
+	const pinConfig = getPinConfig(name, template);
+	if (typeof pinConfig === 'boolean') return DEFAULT_COLOR; // if default pin with no colouring set
+	if (isArray(pinConfig?.color)) return rgbToHex(pinConfig.color); // a custom colour is set, no mapping
+	if (isObject(pinConfig?.color)) return getColorFromMapping(ticket, pinConfig.color); // a custom colour is set with mapping
 	return DEFAULT_COLOR; // if custom pin with no colouring set
 };
 
-export const formatPin = (pinId, position, isSelected: boolean, color: string): IPin => ({
-	id: pinId,
-	position,
-	isSelected,
-	type: 'ticket',
-	colour: hexToGLColor(color),
-});
+export const getPinIcon =  (name: string, template: ITemplate): PinIcon => {
+	const pinConfig = getPinConfig(name, template);
+	if (isObject(pinConfig) && pinConfig.icon) return pinConfig.icon;
+	return 'DEFAULT';
+};
+
+export const getPinId = (propPath, ticketOrId?: ITicket | string) => {
+	const id = (isObject(ticketOrId) ? ticketOrId._id : ticketOrId ) || NEW_TICKET_ID  ;
+	return propPath === DEFAULT_PIN ? id : `${id}.${propPath}`;
+};
+
+// PinIcon is the pin type that comes from the backend
+const pinIconToType = {
+	'DEFAULT' : 'ticket',
+	'ISSUE' : 'issue',
+	'RISK' : 'risk',
+	'MARKER' : 'bookmark',
+};
+
+export const toPin = (propPath: string, template: ITemplate,  ticket: ITicket, isSelected:boolean = false, coordValue?: number[]): IPin => {
+	const colour = hexToGLColor(getPinColorHex(propPath, template, ticket));
+	const icon = getPinIcon(propPath, template);
+	const id = getPinId(propPath, ticket);
+	return {
+		id, 
+		position: (coordValue || get(ticket, propPath) as number[]), 
+		isSelected,
+		type: pinIconToType[icon] as PinType,
+		colour,
+	};
+};
+
+export const toPinIcon = (type: PinType) => (Object.keys(pinIconToType).find((key) => pinIconToType[key] === type) || 'DEFAULT') as PinIcon;
```

Comment:
We have `getPinIcon` and `toPinIcon`. Can we rename them to make it clearer how they get the icon? E.g. (respectively) `getPinIconFromTemplate` and `getPinIconFromType`